timesketch/api/v1/resources/settings.py

```diff
@@ -27,46 +15,78 @@ class SystemSettingsResource(Resource):
+        ):
+            default_provider = next(iter(llm_configs["default"]))
+            try:
+                provider_class = llm_manager.LLMManager.get_provider(default_provider)
+                provider_class(config=llm_configs["default"][default_provider])
+                default_provider_working = True
+            except Exception as e:  # pylint: disable=broad-except
+                logger.debug(
+                    "Default LLM provider '%s' failed to initialize: %s",
+                    default_provider,
+                    str(e),
+                )
+
+        # Initialize feature availability with default status
+        llm_feature_availability = {"default": default_provider_working}
+
+        # Check LLM feature-specific configurations
+        for feature_name, feature_conf in llm_configs.items():
+            if feature_name == "default":
+                continue
+
+            feature_provider_working = False
+            if isinstance(feature_conf, dict) and len(feature_conf) == 1:
+                feature_provider = next(iter(feature_conf))
+                try:
+                    provider_class = llm_manager.LLMManager.get_provider(
+                        feature_provider
+                    )
+                    provider_class(config=feature_conf[feature_provider])
+                    feature_provider_working = True
+                except Exception as e:  # pylint: disable=broad-except
+                    logger.debug(
+                        "LLM provider '%s' for feature '%s' failed: %s",
+                        feature_provider,
+                        feature_name,
+                        str(e),
+                    )
+
+            # Feature is available if either specific provider works
+            # or default provider works
+            llm_feature_availability[feature_name] = (
+                feature_provider_working or default_provider_working
             )
-            result["llm_config_warning"] = warning_message
-            logger.warning(warning_message)
 
-        # Get the search processing timelines setting, default is False if not set.
-        # if set to True, the search processing timelines will be displayed in the UI.
-        search_processing_timelines = current_app.config.get(
-            "SEARCH_PROCESSING_TIMELINES", False
-        )
-        result["SEARCH_PROCESSING_TIMELINES"] = search_processing_timelines
+        result["LLM_FEATURES_AVAILABLE"] = llm_feature_availability
+
+        # TODO(mvd): Remove by 2025/06/01 once all users have updated their config.
+        if current_app.config.get("LLM_PROVIDER") and "default" not in llm_configs:
+            result["llm_config_warning"] = (
+                "Your LLM configuration in timesketch.conf is outdated. "
+                "Please update your LLM_PROVIDER_CONFIGS section to the new format."
+            )
+            logger.warning(result["llm_config_warning"])
```

Comment:
Any reason to wait longer and not remove now? The feature is quite new and with the new config checks the user in the frontend will get the message that something is wrong with their config since the toggles are disabled. So they can go and investigate.

Reply:
Yep I agree that this is a good moment to switch!